### PR TITLE
devdash: init at 0.5.0

### PR DESCRIPTION
**Pull request by h7x4**
## Description of changes

New package:

> Highly Configurable Terminal Dashboard for Developers and Creators 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>devdash</li>
  </ul>
</details>

### CODE REVIEW

File: pkgs/by-name/de/devdash/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchFromGitHub {
+    owner = "Phantas0s";
+    repo = "devdash";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-RUPpgMVl9Cm8uhztdfKnuQ6KdMn9m9PewlT59NnTSiY=";
+  };
+
+  vendorHash = "sha256-xuc8rAkyCInNFxs5itwabqBe4CPg/sAuhcTJsapx7Q8=";
+
+  ldFlags = [
```
Comment (pbsds) [inline pkgs/by-name/de/devdash/package.nix:21]:
```suggestion
  ldflags = [
```

I can't seem to find a single definition using the camelCased version. The manual documents [this](https://nixos.org/manual/nixpkgs/stable/#var-go-ldflags) however. Is it even needed?

**Review by pbsds — CHANGES_REQUESTED** [review]

File: pkgs/by-name/de/devdash/package.nix (hunk 1 above)
Comment (pbsds) [reply pkgs/by-name/de/devdash/package.nix:21]: This lead to #262390

Comment (h7x4) [reply pkgs/by-name/de/devdash/package.nix:21]: Right, seems like a typo on my end. I think I actually might be missing some version flags, so I'll add them and mark as resolved

**Comment by pbsds** [comment]
Tried an exampled config, and it can't open `/bin/cat`

image

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/de/devdash/package.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  ldflags = [
+    "-s -w"
+    "-X github.com/Phantas0s/devdash/cmd.current=${version}"
+    "-X github.com/Phantas0s/devdash/cmd.buildDate=1970-01-01-00:00:01"
+  ];
+
+  patchPhase = ''
+    runHook prePatch
+
+    find . -type f -exec sed -i -e 's|/bin/\([[:alpha:]]\{1,\}\)|/usr/bin/env \1|' {} \;
```
Comment (pbsds) [inline pkgs/by-name/de/devdash/package.nix:31]:
```suggestion
    shopt -s globstar
    substituteInPlace **/*.go --replace '"/bin/' '"/usr/bin/env '
    shopt -u globstar
```

**Review by pbsds — APPROVED** [review]

**Pull request merged**